orbit/cmd/desktop/desktop.go

```diff
@@ -132,19 +137,86 @@ func main() {
 				select {
 				case <-myDeviceItem.ClickedCh:
 					if err := open.Browser(deviceURL.String()); err != nil {
-						log.Printf("open browser my device: %s", err)
+						log.Error().Err(err).Msg("open browser my device")
 					}
 				case <-transparencyItem.ClickedCh:
 					if err := open.Browser("https://fleetdm.com/transparency"); err != nil {
-						log.Printf("open browser transparency: %s", err)
+						log.Error().Err(err).Msg("open browser transparency")
 					}
 				}
 			}
 		}()
 	}
 	onExit := func() {
-		log.Println("exit")
+		log.Info().Msg("exit")
 	}
 
 	systray.Run(onReady, onExit)
 }
+
+// setupLogs configures our logging system to write logs to rolling files and
+// stderr, if for some reason we can't write a log file the logs are still
+// printed to stderr.
+func setupLogs() {
+	stderrOut := zerolog.ConsoleWriter{Out: os.Stderr, TimeFormat: time.RFC3339Nano, NoColor: true}
+
+	dir, err := logDir()
+	if err != nil {
+		log.Logger = log.Output(stderrOut)
+		log.Error().Err(err).Msg("find directory for logs")
+		return
+	}
+
+	logFile := &lumberjack.Logger{
+		Filename:   path.Join(dir, "Fleet", "fleet-desktop.log"),
+		MaxSize:    25, // megabytes
+		MaxBackups: 3,
+		MaxAge:     28, // days
+	}
+
+	log.Logger = log.Output(zerolog.MultiLevelWriter(
+		zerolog.ConsoleWriter{Out: logFile, TimeFormat: time.RFC3339Nano, NoColor: true},
+		stderrOut,
+	))
+}
+
+// logDir returns the default root directory to use for application-level logs.
+//
+// On Unix systems, it returns $XDG_STATE_HOME as specified by
+// https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html if
+// non-empty, else $HOME/.local/state.
+// On Darwin, it returns $HOME/Library/Log.
+// On Windows, it returns %LocalAppData%
+//
+// If the location cannot be determined (for example, $HOME is not defined),
+// then it will return an error.
+func logDir() (string, error) {
```

Comment:
My guess is that we will want utility files for this kind of stuff, but I didn't want to jump the gun until  we have more and we have a better understanding of what would be a good way to organize them.

Having said that, I'm happy to move this somewhere else in this PR.

Reply:
I believe leaving this here for now is a-ok.